web/packages/design/src/Alert/Alert.tsx

```diff
@@ -272,8 +274,6 @@ const AlertIcon = ({
     case 'neutral':
     case 'primary':
       return <Icon.Notification aria-label="Note" {...commonProps} />;
-    default:
-      kind satisfies never;
```

Comment:
If `AlertKind` or `BannerKind` gets extended, `AlertIcon` technically starts returning `JSX.Element | undefined` and no error is being raised. Using `React.FC` doesn't help here as technically `undefined` is a valid thing to return.

Should we also use `noImplicitReturns`? https://www.meticulous.ai/blog/safer-exhaustive-switch-statements-in-typescript

I don't think it's a blocker, it'd probably be better to merge this as is and add `noImplicitReturns` as a followup.

Reply:
Good idea. There was actually only one error, so I enabled it now.